test/lib/flat_mutation_reader_assertions.hh

```diff
@@ -54,13 +54,16 @@ private:
         }
         reset_rts();
     }
-    range_tombstone trim(const range_tombstone& rt, const query::clustering_row_ranges& ck_ranges) const {
-        bound_view::compare less(*_reader.schema());
-        auto ret{rt};
+    bool trim(range_tombstone& rt, const query::clustering_row_ranges& ck_ranges) const {
+        position_in_partition::less_compare less(*_reader.schema());
+        if (ck_ranges.empty()) {
+            return true;
+        }
+        bool valid = false;
```

Comment:
`valid` is misleading here, `trim()` returns whether the rt intersects with the given range. Maybe `intersects` or `relevant` is a better name here.